inst/include/cpp11/logicals.hpp

```diff
@@ -29,22 +29,23 @@ inline SEXP r_vector<r_bool>::valid_type(SEXP data) {
 
 template <>
 inline r_bool r_vector<r_bool>::operator[](const R_xlen_t pos) const {
-  return is_altrep_ ? static_cast<r_bool>(LOGICAL_ELT(data_, pos)) : data_p_[pos];
+  return is_altrep_ ? LOGICAL_ELT(data_, pos) : data_p_[pos];
 }
 
 template <>
-inline r_bool* r_vector<r_bool>::get_p(bool is_altrep, SEXP data) {
+inline typename r_vector<r_bool>::underlying_type* r_vector<r_bool>::get_p(bool is_altrep,
+                                                                           SEXP data) {
   if (is_altrep) {
     return nullptr;
   } else {
-    return reinterpret_cast<r_bool*>(LOGICAL(data));
+    return LOGICAL(data);
```

Comment:
This was the motivating example. `get_p()` now has a return type of `int*` for logicals, which is exactly what `LOGICAL()` returns, so no need for the cast